src/yarp.c

```diff
@@ -2068,7 +2071,33 @@ parse_arguments(yp_parser_t *parser, yp_node_t *arguments, yp_token_type_t termi
       expect(parser, YP_TOKEN_COMMA, "Expected a ',' to delimit arguments.");
     }
 
-    yp_node_t *argument = parse_argument(parser, arguments);
+    if (parsed_block_argument) {
+      yp_error_list_append(&parser->error_list, "Unexpected argument after block argument.", parser->current.start - parser->start);
+    }
+
+    yp_node_t *argument;
+
+    if (accept(parser, YP_TOKEN_STAR)) {
+      if (parsed_double_splat_argument) {
+        yp_error_list_append(&parser->error_list, "Unexpected splat argument after double splat.", parser->current.start - parser->start);
+      }
+      yp_token_t operator = parser->previous;
+      yp_node_t *value = parse_expression(parser, BINDING_POWER_NONE, "Expected to be able to parse an argument.");
```

Comment:
I don't think `BINDING_POWER_NONE` is correct here. That would accept

```ruby
foo(*bar and baz)
```

We should figure out the correct binding power. I've been doing this by manually binary searching through the list of potential values (defined in `yarp.c` `binding_powers` list).